test/WebJobs.Script.Tests.Integration/TestFunctionHost.cs

```diff
@@ -112,6 +112,11 @@ public class TestFunctionHost : IDisposable
 
             var manager = _testServer.Host.Services.GetService<IScriptHostManager>();
             _hostService = manager as WebJobsScriptHostService;
+
+            // Wire up StopApplication calls as they behave in hosted scenarios
+            var lifetime = WebHostServices.GetService<AspNetCore.Hosting.IApplicationLifetime>();
+            lifetime.ApplicationStopping.Register(async () => await _testServer.Host.StopAsync());
```

Comment:
Calling `StopApplication()` in the `TestServer` doesn't work automatically. This makes any call to `StopApplication()` translate to a call to stop the host.